packages/core/src/cache/classes/Contour.ts

```diff
@@ -9,64 +9,137 @@ interface ContourProps {
 }
 
 /**
- * The `Contour` class implements the `IContour` interface and represents a contour in 3D space.
- * It holds information about the contour's id, size in bytes, points, color, and type.
- * The class also provides methods to retrieve the points, color, and type of the contour.
+ * Represents a contour in 3D space.
+ *
+ * @remarks
  * Each Contour is part of a ContourSet, and each ContourSet is part of a Geometry.
+ * It holds information about the contour's id, size in bytes, points, color, and type.
  */
 export class Contour {
+  /** Unique identifier for the contour */
   readonly id: string;
+
+  /** Size of the contour data in bytes */
   readonly sizeInBytes: number;
-  points: Point3[];
-  color: Point3;
-  type: ContourType;
-  segmentIndex: number;
 
+  /** Array of 3D points defining the contour */
+  private _points: Point3[];
```

Comment:
Should you use a PointsManager - the efficiency of the contour access is better because you can just get references to particular points, and the size is just the native data type times the number of values total?  Not sure this is worth doing right now, but I thought I'd mention it.

Reply:
Great mention, i added a todo